src/main/java/org/craftercms/studio/impl/v2/service/clipboard/internal/ClipboardServiceInternalImpl.java

```diff
@@ -0,0 +1,109 @@
+import org.craftercms.studio.api.v1.service.workflow.WorkflowService;
+import org.craftercms.studio.api.v2.service.clipboard.internal.ClipboardServiceInternal;
+import org.craftercms.studio.model.clipboard.Operation;
+import org.craftercms.studio.model.clipboard.PasteItem;
+
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+
+import static org.apache.commons.collections4.CollectionUtils.isNotEmpty;
+import static org.apache.commons.io.FilenameUtils.getFullPathNoEndSeparator;
+import static org.apache.commons.io.FilenameUtils.getName;
+
+/**
+ * Default implementation of {@link ClipboardServiceInternal}
+ *
+ * <p>Note: This class could be removed in the future if the logic is moved to the new content service</p>
+ *
+ * @author joseross
+ * @since 3.2
+ */
+public class ClipboardServiceInternalImpl implements ClipboardServiceInternal {
+
+    private static final Logger logger = LoggerFactory.getLogger(ClipboardServiceInternalImpl.class);
+
+    protected ContentService contentService;
+    protected WorkflowService workflowService;
+
+    public ClipboardServiceInternalImpl(ContentService contentService, WorkflowService workflowService) {
+        this.contentService = contentService;
+        this.workflowService = workflowService;
+    }
+
+    public List<String> pasteItems(String siteId, Operation operation, String targetPath, PasteItem item)
+            throws ServiceLayerException, UserNotFoundException {
+        var pastedItems = new LinkedList<String>();
+        pasteItemsInternal(siteId, operation, targetPath, List.of(item), pastedItems);
+        return pastedItems;
+    }
+
+    // Code based on the original clipboard service v1
+    protected void pasteItemsInternal(String siteId, Operation operation, String targetPath, List<PasteItem> items,
+                                      List<String> pastedItems) throws ServiceLayerException, UserNotFoundException {
+        for(var item : items) {
+            try {
+                String newPath = null;
+                switch (operation) {
+                    case CUT:
+                        // RDTMP_COPYPASTE
+                        // CopyContent interface is able to send status and new path yet
+                        workflowService.cleanWorkflow(item.getPath(), siteId, Collections.emptySet());
+                        newPath = contentService.moveContent(siteId, item.getPath(), targetPath);
+                        break;
+                    case COPY:
+                        // RDTMP_COPYPASTE
+                        // CopyContent interface is able to send status and new path yet
+                        newPath = contentService.copyContent(siteId, item.getPath(), targetPath);
+
+                        // recurse on copied children
+                        if (isNotEmpty(item.getChildren())) {
+                            pasteItemsInternal(siteId, operation, newPath, item.getChildren(), pastedItems);
```

Comment:
Nothing is returned, therefore the children's paths are not being returned.

Reply:
The children's path are added to the list (last parameter of the method)